### PR TITLE
build: Fix compilation error with gleam 0.23

### PR DESCRIPTION
**Pull request by Blond11516**
The project currently fails to compile with gleam 0.23 with the following error:
```
error: Duplicate import
  ┌─ build/packages/gleam_stdlib/src/gleam/dynamic.gleam:3:8
  │
3 │ import gleam/map
  │        ^^^^^^^^^ First imported here
  ·
8 │ import gleam/map.{Map}
  │        ^^^^^^^^^^^^^^^ Reimported here

map has been imported multiple times.
Names in a Gleam module must be unique so one will need to be renamed.
```

Updating `gleam_stdlib` to 0.23 fixes the issue.

I'm not sure if bumping up the requirement like that is desirable for a library like gleam_http, so feel free to close this if it isn't!

### CODE REVIEW

File: gleam.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ links = [
 ]
 
 [dependencies]
-gleam_stdlib = "~> 0.18"
+gleam_stdlib = "~> 0.23"
```
Comment (lpil) [inline gleam.toml:13]: Could you revert this please? This would prevent older stdlib users from using future `gleam_http` releases.

**Review by lpil — COMMENTED** [review]
Thanks for the patch!

`manifest.toml` isn't used for libaries so we could probably just gitignore it to prevent this from happening in future if people clone this repo.

File: gleam.toml (hunk 1 above)
Comment (Blond11516) [reply gleam.toml]: Done! And I've ignored `manifest.toml` as well.

**Review by lpil — APPROVED** [review]
Thank you

**Comment by lpil** [comment]
Sorry, could you fix the merge conflict please? 🙏

**Comment by Blond11516** [comment]
Oh that's my bad, I didn't notice it! Should be good now.

**Comment by lpil** [comment]
Thank you

**Pull request merged**